src/gravitational_waves/gw.cpp

```diff
@@ -220,61 +335,412 @@ Nintegrate_SNR(GravitationalWave &obj, const double fmin, const double fmax)
+
+    if (almin > al) return 0;
+
+    std::vector<std::vector<double>> iv = {{vp, almin}, {0, almax}};
+    while (abs(iv[1][0] - iv[0][0]) > 1.e-7)
+    {
+      vpm        = (iv[1][0] + iv[0][0]) / 2.0;
+      double alm = getalNwow(vpm, vm, vw, cs2b, cs2s).first;
+      if (alm > al)
+      {
+        iv = {iv[0], {vpm, alm}};
+      }
+      else
+      {
+        iv = {{vpm, alm}, iv[1]};
+      }
+    }
+    vp                 = (iv[1][0] + iv[0][0]) / 2.0;
+    std::tie(Ksh, wow) = getKandWow(vw, mu(vw, vp), cs2s, vprofile_false);
+  }
+  double Krf = 0;
+  if (mode > 0) // Not deflagration
+  {
+    auto [Krf_val, wow3] = getKandWow(vw, mu(vw, vm), cs2b, vprofile_true);
+    Krf                  = -wow * getwow(vp, vm) * Krf_val;
+  }
+  vprofile = vprofile_true;
+  std::reverse(vprofile.begin(), vprofile.end());
+  vprofile.insert(vprofile.end(), vprofile_false.begin(), vprofile_false.end());
+  return (Ksh + Krf) / al;
 }
 
+double Getkappa_col(const double &Tstar,
+                    const int &pnlo_scaling,
+                    const double &HR,
+                    BounceSolution &BACalc)
+{
+  // Calculate the false and true vacuum
+  const std::vector<double> FalseVacuum = BACalc.modelPointer->MinimizeOrderVEV(
+      BACalc.phase_pair.false_phase.Get(Tstar).point);
+  const std::vector<double> TrueVacuum = BACalc.modelPointer->MinimizeOrderVEV(
+      BACalc.phase_pair.true_phase.Get(Tstar).point);
+  // Potential difference
+  const double dV = BACalc.modelPointer->VEff(FalseVacuum, Tstar) -
+                    BACalc.modelPointer->VEff(TrueVacuum, Tstar);
+  // Potential energy contribution to the energy of the initial bubble
+  const double E0V = BACalc.GetBounceSol(Tstar) * Tstar / 2.;
+  // Initial bubble radius
+  double R0 = pow(3 * E0V / (4 * M_PI * dV), 1. / 3.);
+
+  // Mean bubble seperation
+  const double Rstar = HR / BACalc.HubbleRate(Tstar);
+  // Energy densitity radiation dominated Universe
+  const double RhoGamma = BACalc.CalculateRhoGamma(Tstar);
+
+  // Calculate masses squared in false and true vacuum
+  std::vector<double> HiggsSq_False =
+      BACalc.modelPointer->HiggsMassesSquared(FalseVacuum, Tstar);
+  std::vector<double> LeptonSq_False =
+      BACalc.modelPointer->LeptonMassesSquared(FalseVacuum);
+  std::vector<double> QuarkSq_False =
+      BACalc.modelPointer->QuarkMassesSquared(FalseVacuum);
+  std::vector<double> GaugeSq_False =
+      BACalc.modelPointer->GaugeMassesSquared(FalseVacuum, Tstar);
+
+  std::vector<double> HiggsSq_True =
+      BACalc.modelPointer->HiggsMassesSquared(TrueVacuum, Tstar);
+  std::vector<double> LeptonSq_True =
+      BACalc.modelPointer->LeptonMassesSquared(TrueVacuum);
+  std::vector<double> QuarkSq_True =
+      BACalc.modelPointer->QuarkMassesSquared(TrueVacuum);
+  std::vector<double> GaugeSq_True =
+      BACalc.modelPointer->GaugeMassesSquared(TrueVacuum, Tstar);
+
+  double P_LO = 0; // Pressure at LO. 1 -> 1
+
+  for (auto massSq : HiggsSq_True)
```

Comment:
```suggestion
  for (const auto& massSq : HiggsSq_True)
```

Otherwise you are making a lot of copies

Reply:
Updated in new branch